## *Programs installed locally on Maria's account:* :smile:
##R packages:
Using R version 3.2.2 installed locally:
```export PATH=/home/armita/prog/R/R-3.2.2/bin:${PATH}```
and libraries stored in ```/home/sobczm/R/x86_64-pc-linux-gnu-library/3.2```

[ade4] (https://cran.r-project.org/web/packages/ade4/index.html): Analysis of Ecological Data : Exploratory and Euclidean Methods in Environmental Sciences

[adegenet] (http://adegenet.r-forge.r-project.org/): a R package for the multivariate analysis of genetic markers

[ggplot2] (http://ggplot2.org/): graphing package implemented on top of the R statistical package

[pegas] (https://cran.r-project.org/web/packages/pegas/index.html): Population and Evolutionary Genetics Analysis System

[PopGenome] (https://cran.r-project.org/web/packages/PopGenome/index.html): An Efficient Swiss Army Knife for Population Genomic Analyses

[poppr] (http://grunwaldlab.cgrb.oregonstate.edu/poppr-r-package-population-genetics): Genetic Analysis of Populations with Mixed Reproduction

[vcfR] (https://cran.r-project.org/web/packages/vcfR/index.html): Manipulate and Visualize VCF Data

[WhopGenome] (https://cran.r-project.org/web/packages/WhopGenome/index.html): High-Speed Processing of VCF, FASTA and Alignment Data

##Standalone
[ABRA ver. 0.97] (https://github.com/mozack/abra): Improved coding indel detection via assembly-based
realignment `/home/sobczm/bin/abra/bin`

[BayeScan ver. 2.1] (http://cmpg.unibe.ch/software/BayeScan/): detecting natural selection from population-based genetic data
`/home/sobczm/bin/bayescan2.1/binaries/bayescan_2.1`

**BEAST requires Java 8 - downloaded it to a local directory and change path for default Java in my profile:**
```
export JAVA_HOME=/home/sobczm/bin/jre1.8.0_101
export PATH="$JAVA_HOME/bin:$PATH"
```
[BEAST ver. 1.8.3] (http://beast.bio.ed.ac.uk/) package - Bayesian analysis of molecular sequences using MCMC. Includes: BEAST, BEAUti, LogCombiner, TreeAnnotator.  `/home/sobczm/bin/beast/BEASTv1.8.3/bin`

[BEAST ver. 2.4.2] (http://beast2.org/) package. Includes: BEAST, BEAUti, LogCombiner, TreeAnnotator, DensiTree. `/home/sobczm/bin/beast/BEASTv2.4.2/bin` 

[BEASTGen ver. 1.0.2] (http://beast.bio.ed.ac.uk/beastgen) Creates BEAST XML input files. 
`/home/sobczm/bin/beast/BEASTGenv1.0.2/bin`

[BUSCO ver 1.22] (http://busco.ezlab.org/): Assessing genome assembly and annotation completeness with Benchmarking Universal Single-Copy Orthologs
`/home/sobczm/bin/BUSCO_v1.22/BUSCO_v1.22.py`

**Dependencies**:
```
export PATH=/home/sobczm/bin/hmmer-3.1b2/binaries:${PATH}
export PATH=/home/armita/prog/python3/Python-3.3.5/bin:${PATH}
export PATH=/home/armita/prog/ncbi-rmblastn-2.2.28/bin:${PATH}
export AUGUSTUS_CONFIG_PATH=/home/armita/prog/augustus-3.1/config
export PATH=/home/armita/prog/augustus-3.1/bin:${PATH}
export PATH=/home/armita/prog/augustus-3.1/scripts:${PATH}
export PATH=/home/armita/prog/emboss/EMBOSS-4.0.0/bin:${PATH}
```
[DendroPy ver. 4.1.0] (https://pythonhosted.org/DendroPy/) Python library for phylogenetic computing. `/home/sobczm/bin/DendroPy` *Need to export it to PYTHONPATH

[EIGENSOFT ver. 6.1.2] (https://genetics.med.harvard.edu/reich/Reich_Lab/Software.html) A set of population structure dectection methods. `/home/sobczm/bin/EIG/bin`

[FastTree ver. 2.1.9] (http://www.microbesonline.org/fasttree/) Approximately Maximum-Likelihood Trees for Large Alignments `/home/sobczm/bin/FastTree2.1.9`  

[FigTree ver. 1.4.2] (http://tree.bio.ed.ac.uk/software/figtree/) Viewing of phylogenetic trees and production of publication-ready figures. `/home/sobczm/bin/FigTree_v1.4.2/bin`

[freebayes ver. v1.0.2] (https://github.com/ekg/freebayes) Bayesian haplotype-based polymorphism discovery and genotyping. `/home/sobczm/bin/freebayes/bin`

[GATK ver. 3.6] (https://software.broadinstitute.org/gatk/) Genome Analysis Toolkit - Variant Discovery in High-Throughput Sequencing Data. `/home/sobczm/bin/GenomeAnalysisTK-3.6`

[MAFFT ver. 7.222] (http://mafft.cbrc.jp/alignment/software/) Rapid multiple sequence alignment based on fast Fourier transform
`/home/sobczm/bin/mafft-7.222/bin`

[PAL2NAL] (http://www.bork.embl.de/pal2nal/): Robust conversion of protein sequence alignments into the corresponding codon alignments/home. `/sobczm/bin/pal2nal.v14`

[PAML ver. 4.8] (http://abacus.gene.ucl.ac.uk/software/paml.html) A package of programs for phylogenetic analyses of DNA or protein sequences using maximum likelihood. `/home/sobczm/bin/paml4.8/bin`

[PartitionFinder ver. 1.1.1] (http://www.robertlanfear.com/partitionfinder/) A program to select best-fit partitioning schemes and models of molecular evolution for phylogenetic analyses `/home/sobczm/bin/PartitionFinder1.1.1` *Needs to be run with Anaconda Python distribution installed in `/home/sobczm/bin/anaconda2/bin`

[PicardTools ver. 2.5.0] (https://broadinstitute.github.io/picard/) A set of command line tools for manipulating formats such as SAM/BAM/CRAM and VCF. `/home/sobczm/bin/picard-tools-2.5.0`

[PhyloNet ver. 3.5.5] (http://bioinfo.cs.rice.edu/phylonet) & [PhyloNetHMM ver. 0.1] (http://bioinfo.cs.rice.edu/software/phmm) Bayesian inference of reticulate phylogenies under the multispecies network coalescent `/home/sobczm/bin/phmm`

[RAxML ver. 8.2.9] (http://sco.h-its.org/exelixis/web/software/raxml/index.html) a ML a tool for phylogenetic analysis and post-analysis of large phylogenies `/home/sobczm/bin/RAxML8.2.9`

[SplitsTree ver. 4] (http://www.splitstree.org/) Program for computing unrooted phylogenetic networks from molecular sequence data. `/home/sobczm/bin/splitstree4` 

[STRUCTURE ver. 2.3.4] (http://pritchardlab.stanford.edu/structure.html) A package for using multi-locus genotype data to investigate population structure `/home/sobczm/bin/structure`

[structure Harvester ver. 0.6.93] (http://taylor0.biology.ucla.edu/structureHarvester) Downstream processing of STRUCTURE results to calculate Evanno’s Δk value and prepares input file for CLUMPP `/home/sobczm/bin/structureHarvester`

[CLUMPP ver. 1.1.2] (https://web.stanford.edu/group/rosenberglab/clumpp.html) Permutes the clusters output by independent runs of STRUCTURE, so that they match up as closely as possible. `/home/sobczm/bin/CLUMPP_Linux64.1.1.2`

[distruct ver. 1.1] (https://web.stanford.edu/group/rosenberglab/distruct.html) A program to graphically display results produced by STRUCTURE or by other similar programs. `/home/sobczm/bin/distruct1.1`

[Stampy ver. 1.0.29] (http://www.well.ox.ac.uk/project-stampy) Sensitive mapping of Illumina reads. `/home/sobczm/bin/stampy-1.0.29`

[Tracer ver. 1.6] (http://tree.bio.ed.ac.uk/software/tracer/) A program for analysing the trace files generated by Bayesian MCMC runs.
`/home/sobczm/bin/beast/Tracer_v1.6/bin`

[Treemix ver. 1.12] (https://bitbucket.org/nygcresearch/treemix/wiki/Home): estimation of population trees with admixture.
`/home/sobczm/bin/treemix-1.12` 

[vcflib] (https://github.com/vcflib/vcflib): a simple C++ library for parsing and manipulating VCF files, + many command-line utilities. `/home/sobczm/bin/vcflib/bin`

[VCFtools] (https://vcftools.github.io): another set of C++ and Perl libraries for analysing VCF files. 
```
/home/sobczm/bin/vcftools/bin  
#SET PERL PATH
export PERL5LIB=/home/sobczm/bin/vcftools/share/perl/5.14.2
```
[PyVCF] (https://github.com/jamescasbon/PyVCF) A VCF v. 4.0 and 4.1 parser for Python.
```
/home/sobczm/bin/PyVCF/scripts
#SET PYTHON PATH
export PYTHONPATH="$PYTHONPATH:/home/sobczm/bin/PyVCF/lib/python2.7/site-packages/"
```



