## List of bash pipelines used in the analysis of variants in the ananassa genome.
### Basic toy analysis of individual genotypes.
[explore_ananassa_sql_snpchip.sh](https://github.com/harrisonlab/popgen/blob/master/snp/explore_ananassa_sql_snpchip.sh)

### Extended analysis of individual genotypes
[analyse_snp_chip_genotypes.sh](https://github.com/harrisonlab/popgen/blob/master/snp/analyse_snp_chip_genotypes.sh)

[ananassa_nuc_div_by_chromosome.sh](https://github.com/harrisonlab/popgen/blob/master/snp/ananassa_nuc_div_by_chromosome.sh)

### Basic toy analysis of individual haplotypes.
[insert_ananassa_haplotypes.sh](https://github.com/harrisonlab/popgen/blob/master/snp/insert_ananassa_haplotypes.sh)

### Comparison of map-based and Impute-based haplotyping
[comparison_haplotyping_procedures.sh](https://github.com/harrisonlab/popgen/blob/master/snp/comparison_haplotyping_procedures.sh)

### Haplotyping of EMxFE, FLxCH, HAxRG EMR mapping populations with shapeit
[mapping_pops_haplotyping.sh](https://github.com/harrisonlab/popgen/blob/master/snp/mapping_pops_haplotyping.sh)

### Troubleshooting haplotyping carried out in the file above.
[troubleshooting_haplotyping.sh](https://github.com/harrisonlab/popgen/blob/master/snp/troubleshooting_haplotyping.sh)

### Testing shapeit settings to improve haplotyping
[improve_haplotyping.sh](https://github.com/harrisonlab/popgen/blob/master/snp/improve_haplotyping.sh)
### Testing shapeit settings to improve haplotyping 2
[improve_haplotyping2.sh](https://github.com/harrisonlab/popgen/blob/master/snp/improve_haplotyping2.sh)
### Testing shapeit settings to improve haplotyping based on vcf input.
[improve_haplotyping3.sh](https://github.com/harrisonlab/popgen/blob/master/snp/improve_haplotyping3.sh)
### Testing shapeit settings to improve haplotyping based on vcf input.
[improve_haplotyping4.sh](https://github.com/harrisonlab/popgen/blob/master/snp/improve_haplotyping4.sh)
### Testing shapeit settings to improve haplotyping based on vcf input. Testing using haplotypes from another population to improve phasing.
[improve_haplotyping5.sh](https://github.com/harrisonlab/popgen/blob/master/snp/improve_haplotyping5.sh)
### Extended analysis of individual genotypes 2, following fixing of mislabelled samples.
[analyse_snp_chip_genotypes2.sh](https://github.com/harrisonlab/popgen/blob/master/snp/analyse_snp_chip_genotypes2.sh)
### Local genotype association study for 10-20 markers around major QTLs for crown rot.
[crown_rot_qtl_markers_local_1.sh](https://github.com/harrisonlab/popgen/blob/master/snp/crown_rot_qtl_markers_local_1.sh)
### GWAS for crown rot using PLINK and TASSEL (GUI)
[crown_rot_qtl_markers_gwas_1.sh](https://github.com/harrisonlab/popgen/blob/master/snp/crown_rot_qtl_markers_gwas_1.sh)
### Re-analysis of GWAS and SBC genotypes focussing on plate provenance
[crown_rot_qtl_markers_gwas_plate.sh](https://github.com/harrisonlab/popgen/blob/master/snp/crown_rot_qtl_markers_gwas_plate.sh)
### Various checks of the ananassa snp chip database - QC. 
[ananassa_db_qc.sh](https://github.com/harrisonlab/popgen/blob/master/snp/ananassa_db_qc.sh)
### GWAS analysis of everbearer trait
[everbearer_gwas.sh](https://github.com/harrisonlab/popgen/blob/master/snp/everbearer_gwas.sh)
